android/app/src/main/java/com/microsoft/codepush/react/CodePushNativeModule.java

```diff
@@ -0,0 +1,482 @@
+    private CodePushTelemetryManager codePushTelemetryManager;
+    private SettingsManager settingsManager;
+
+    private static final String REACT_APPLICATION_CLASS_NAME = "com.facebook.react.ReactApplication";
+    private static final String REACT_NATIVE_HOST_CLASS_NAME = "com.facebook.react.ReactNativeHost";
+
+    public CodePushNativeModule(ReactApplicationContext reactContext, CodePush codePush, CodePushPackage codePushPackage, CodePushTelemetryManager codePushTelemetryManager, SettingsManager settingsManager) {
+        super(reactContext);
+        this.codePush = codePush;
+        this.codePushPackage = codePushPackage;
+        this.codePushTelemetryManager = codePushTelemetryManager;
+        this.settingsManager = settingsManager;
+    }
+
+    @Override
+    public Map<String, Object> getConstants() {
+        final Map<String, Object> constants = new HashMap<>();
+
+        constants.put("codePushInstallModeImmediate", CodePushInstallMode.IMMEDIATE.getValue());
+        constants.put("codePushInstallModeOnNextRestart", CodePushInstallMode.ON_NEXT_RESTART.getValue());
+        constants.put("codePushInstallModeOnNextResume", CodePushInstallMode.ON_NEXT_RESUME.getValue());
+
+        constants.put("codePushUpdateStateRunning", CodePushUpdateState.RUNNING.getValue());
+        constants.put("codePushUpdateStatePending", CodePushUpdateState.PENDING.getValue());
+        constants.put("codePushUpdateStateLatest", CodePushUpdateState.LATEST.getValue());
+
+        return constants;
+    }
+
+    @Override
+    public String getName() {
+        return "CodePush";
+    }
+
+    private Class tryGetClass(String className) {
+        try {
+            return Class.forName(className);
+        } catch (ClassNotFoundException e) {
+            return null;
+        }
+    }
+
+    private boolean isReactApplication(Context context) {
+        Class reactApplicationClass = tryGetClass(REACT_APPLICATION_CLASS_NAME);
+        if (reactApplicationClass != null && reactApplicationClass.isInstance(context)) {
+            return true;
+        }
+
+        return false;
+    }
+
+    private void loadBundleLegacy() {
+        Activity currentActivity = getCurrentActivity();
+        Intent intent = currentActivity.getIntent();
+        currentActivity.finish();
+        currentActivity.startActivity(intent);
+
+        codePush.invalidateCurrentInstance();
+    }
+
+    private void loadBundle() {
+        codePush.clearDebugCacheIfNeeded();
+        Activity currentActivity = getCurrentActivity();
+
+        if (!ReactActivity.class.isInstance(currentActivity)) {
+            // Our preferred reload logic relies on the user's Activity inheriting
+            // from the core ReactActivity class, so if it doesn't, we fallback
+            // early to our legacy behavior.
+            loadBundleLegacy();
+        } else {
+            try {
+                ReactActivity reactActivity = (ReactActivity)getCurrentActivity();
```

Comment:
Can't you cast "currentActivity" instead of needing to retrieve the current activity again?